main.go

```diff
@@ -118,9 +126,30 @@ func prefetch(url string) (string, error) {
 	contents, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		log.Print("Prefetch failed reading body for: ", url, err)
+	}
+	dname, err := os.MkdirTemp("", "")
+	if err != nil {
+		log.Fatal(err)
+	}
+	buffer := bytes.NewBuffer(contents)
+	extract.Archive(context.Background(), buffer, dname, nil)
+	entries, err := os.ReadDir(dname)
+	if err != nil {
 		return "", err
 	}
-	sha256 := fmt.Sprintf("%x", sha256.Sum256(contents))
+	var entryPath string
+	for _, entry := range entries {
+		if entry.IsDir() {
+			entryPath = filepath.Join(dname, entry.Name())
+			break
+		}
+	}
+	err = nar.DumpPath(h, entryPath)
+	if err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(dname)
+	sha256 = nixbase32.EncodeToString(h.Sum(nil))
```

Comment:
Why not remove the `var sha256 string` above and make this a

```suggestion
	return nixbase32.EncodeToString(h.Sum(nil)), nil
```